client/state/navigation/middleware.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { NAVIGATE } from 'state/action-types';
+
+export const navigationMiddleware = () => {
+	return ( next ) => ( action ) => {
+		if ( action.type === NAVIGATE && action.path ) {
+			page( action.path );
+		}
+
+		next( action );
```

Comment:
Dispatches can have a return value, so it's a good idea to `return next( action )` here. We don't use this feature much at the moment, but let's keep the middleware chain compliant anyway.

Other than that, this patch looks nice :)

Reply:
👍